services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/handlers/DecodeUrlEncodedResponseInterceptor.java

```diff
@@ -169,4 +171,13 @@ private static List<MultipartUpload> decodeMultipartUpload(List<MultipartUpload>
                                                             .map(u -> u.toBuilder().key(urlDecode(u.key())).build())
                                                             .collect(Collectors.toList()));
     }
+
+    private static List<DeleteMarkerEntry> decodeDeleteMarkers(List<DeleteMarkerEntry> entries) {
+        if (entries == null) {
```

Comment:
nit : want to check for empty list too ?

Reply:
other list method don't (`decodeCommonPrefixes(...)`, `decodeMultipartUpload(...)`) I'd rather keep similar behaviour between all of them.

Reply:
Sounds reasonable. Looks like we would return an empty list if it is empty. Is that intended behaviour ?

Reply:
yep